### PR TITLE
[TF-TRT]  Refactor Softmax converter to use OpConverterBase

### PR DESCRIPTION
**Pull request by pavanimajety**
This commit refactors the Softmax TF-TRT conversion to use OpConverterBase classes.

### CODE REVIEW

**Comment by bixia1** [comment]
The dependent PR is merged, can you rebase?

File: tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc:1]: Please add license header, can copy and paste from another source file then change the year to 2022.

File: tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    ITensorProxyPtr logits_tensor = inputs.at(0).tensor();
+    const int num_trt_dims = logits_tensor->getDimensions().nbDims;
+
+    // Perform Softmax operation:
+    nvinfer1::ISoftMaxLayer *layer =
+        params.converter->network()->addSoftMax(*logits_tensor->trt_tensor());
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
+    params.converter->SetLayerName(layer, node_def);
+    // Tensorflow SoftMax assumes applying softmax on the last dimension.
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc:52]: s/assumes applying/applies/

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — CHANGES_REQUESTED** [review]
compilation errors:
[third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:105](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc?l=105&ws=tap-prod-presubmit/207445590&snapshot=2):11: error: enumeration value 'kSOFTMAX' not handled in switch [-Werror,-Wswitch]
  switch (layer_type) {
          ^~~~~~~~~~
1 error generated.

[third_party/tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc:43](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc?l=43&ws=tap-prod-presubmit/207445590&snapshot=2):17: error: unused variable 'node_def' [-Werror,-Wunused-variable]
    const auto &node_def = params.node_def;
                ^
1 error generated.

**Review by bixia1 — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @pavanimajety Can you please resolve conflicts? Thank you!

**Review by bixia1 — COMMENTED** [review]
Would you please squash the commits?

**Review by bixia1 — APPROVED** [review]
